test/lit/passes/reorder-globals.wast

```diff
@@ -0,0 +1,404 @@
+;; NOTE: Assertions have been generated by update_lit_checks.py and should not be edited.
+;; RUN: foreach %s %t wasm-opt --reorder-globals-always -S -o - | filecheck %s
+;; RUN: foreach %s %t wasm-opt --reorder-globals-always --roundtrip -S -o - | filecheck %s --check-prefix RTRIP
```

Comment:
Unless I missed something, the roundtrip output is the same as the normal output, so there's no need for a separate prefix.

```suggestion
;; RUN: foreach %s %t wasm-opt --reorder-globals-always --roundtrip -S -o - | filecheck %s
```

Reply:
The output is identical, but I was thinking this might prevent us from somehow emitting invalid binaries. Like if we had an invariant on the order in the IR (which apparently we don't, as mentioned above, but if we did, and we broke it).

(see comment on lines 5-6)

Reply:
Yeah, testing roundtripping sounds good, but it can reuse the same `CHECK` output as the previous `RUN` line.

Reply:
Oh, ok, makes sense.